### PR TITLE
Guardbuddies can now use guns

### PR DESCRIPTION
**Pull request by Superlagg**
[FEATURE][INPUT WANTED]

## About the PR 

Guardbuddies can now use guns! You can also swap out their module, or remove it altogether!

To mess with a Buddy's module, unlock it with a proper ID, then click on them while holding a tool module. 

While they're unlocked, you can use a crowbar on them to pry out their tool module. This leaves the Buddy without any way to defend itself, and thus won't appreciate people picking on it.

To give the Buddy a gun, you need to remove its module, then, while it's still unlocked, click on them with a gun. Note that, unless you emag or emp them, they'll only accept a narrow range of non-lethal guns. Then, to prevent people from just crowbarring the gun back off them, you can relock the Buddy.

Energy weapons draw from the Buddy's internal cell, whereas kinetic weapons just use their own ammunition. 

There's also a traitor RD module that'll allow the Buddy to reload kinetic guns using its internal cell, drawing power based on how much damage the ammunition does, replenishing the magazine once it runs out. The module can be scanned by a traitor scanner thing, so if you want to work with badguy mechanics and make an army of shotgun brickboys, all power to you.

Also, Buddies don't need a tool when you're building them. In fact, you can't add a tool until after they're built!
...might make it tricky for anyone other than the RD to build one. Maybe I'll have them start life as unlocked.

## Why's this needed? 

I want Harner to shoot me something other than that dumb smirk.

## Changelog

```
(u)Superlagg:
(*)Guardbuddies can now use guns!
(*)Traitor RDs can order an ammo fabricator for their Guardbuddies!
(+)You can also swap out or remove Guardbuddy tools on the fly!
```

### CODE REVIEW

**Comment by Tarmunora** [comment]
Having the option still to add a tool before construction would help with not making it hard for nonRDs to make buddies, One other thing to check is to make sure the buddy factory in debris can still be used to make buddies

**Comment by ZeWaka** [comment]
Ammo fab major change

**Comment by Superlagg** [comment]
Newly build Buddies now start unlocked, allowing anybody to give it a module or a gun. Course, giving them a gun without them immediately arresting you for having a gun is a bit trickier.

Also upgraded ammofab changelog thing to major.

File: code/modules/robotics/bot/guardbot.dm [hunk 1]
```diff
@@ -3150,9 +3395,9 @@
 			if(!created_cell || stage != 2)
 				boutput(user, "<span class='alert'>You need to add a power cell first!</span>")
 				return
-			if(!created_module)
-				boutput(user, "<span class='alert'>You need to add a tool module first!</span>")
-				return
+//			if(!created_module)
+//				boutput(user, "<span class='alert'>You need to add a tool module first!</span>")
+//				return
```
Comment (ZeWaka) [inline code/modules/robotics/bot/guardbot.dm:3400]:
```suggestion
```

File: code/modules/robotics/bot/guardbot.dm [hunk 2]
```diff
@@ -3173,12 +3418,12 @@
...
-
-			W.set_loc(src)
-			src.created_module = W
-			boutput(user, "You add the [W.name] to [src]!")
+//		else if((istype(W, /obj/item/device/guardbot_tool)) && stage == 2 && !created_module)
+//			user.drop_item()
+//
+//			W.set_loc(src)
+//			src.created_module = W
+//			boutput(user, "You add the [W.name] to [src]!")
```
Comment (ZeWaka) [inline code/modules/robotics/bot/guardbot.dm:3426]:
```suggestion
```

File: code/modules/robotics/bot/guardbot.dm [hunk 3]
```diff
@@ -3902,9 +4166,9 @@
 			if(!created_cell || stage != 2)
 				boutput(user, "<span class='alert'>You need to add a power cell first!</span>")
 				return
-			if(!created_module)
-				boutput(user, "<span class='alert'>You need to add a tool module first!</span>")
-				return
+//			if(!created_module)
+//				boutput(user, "<span class='alert'>You need to add a tool module first!</span>")
+//				return
```
Comment (ZeWaka) [inline code/modules/robotics/bot/guardbot.dm:4171]:
```suggestion
```

File: code/modules/robotics/bot/guardbot.dm [hunk 4]
```diff
@@ -3921,16 +4185,16 @@
...
-
-			W.set_loc(src)
-			src.created_module = W
-			boutput(user, "You add the [W.name] to [src]!")
+//		else if((istype(W, /obj/item/device/guardbot_tool)) && stage == 2 && !created_module)
+//			user.drop_item()
+//
+//			W.set_loc(src)
+//			src.created_module = W
+//			boutput(user, "You add the [W.name] to [src]!")
```
Comment (ZeWaka) [inline code/modules/robotics/bot/guardbot.dm:4197]:
```suggestion
```

**Pull request merged**